collections-contrib/src/test/scala/strawman/collection/decorators/SeqDecoratorTest.scala

```diff
@@ -17,4 +18,20 @@ class SeqDecoratorTest {
     assertEquals(List(0, 1, 2), List(1).intersperse(0, 5, 2))
   }
 
+  // This test just checks that there is no compilation error
+  @Test def genericDecorator(): Unit = {
+    val list = List(1, 2, 3)
+    val vector = Vector(1, 2, 3)
+    val array = Array(1, 2, 3)
+    val string = "foo"
+    list.intersperse(0)
+    list.view.intersperse(0)
+    vector.intersperse(0)
+    vector.view.intersperse(0)
+    array.intersperse(0)
+    array.view.intersperse(0)
+    string.intersperse(' ')
+    string.view.intersperse(' ')
```

Comment:
This approach works with `List`, `Vector` but also views, `Array` and `String`.

However, there seems to be some type information lost somewhere:

~~~ scala
    val result = list.intersperse(0)
    result: List[Int]
~~~